url3f8d6440dd_huiwq1990_CognativeModel_tgz-pJ8-clarion_samples_HelloWorldJ8/src/clarion/samples/HelloWorld.java

```diff
@@ -0,0 +1,399 @@
+		/*Set the (optional) learning rate and momentum parameters.*/
+		//acsbp.MOMENTUM = .5;
+		//acsbp.LEARNING_RATE = .2;
+		 
+		//Add the neural net to the bottom level of the ACS.
+		acs.addBLModule(acsbp);
+	}
+	
+	/**
+	 * This method runs the agent through the Hello World task.
+	 */
+	public void run(CLARION Agent) {
+		//For keeping track of the feedback.
+		int feedback = 0;
+		//For keeping track of the chosen action.
+		AbstractAction ChosenAction;
+		//Run the task for the specified number of trials.
+		for(int i = 0; i < NumberTrials; i++)
+		{
+			//Randomly choose a salutation to give.
+			if(Math.random() > .5)
+			{
+				//Specify the salutation to give as "Hello".
+				CombinedInputSpace.get("Salutations").get("Hello").setActivation(1);
+				CombinedInputSpace.get("Salutations").get("Goodbye").setActivation(0);
+				//Specify the stimulus for the AFFILIATON_AND BELONGINNESS drive.
+				CombinedInputSpace.get(Personality.PrimaryDrives.AFFILIATION_AND_BELONGINGNESS)
+					.get(Drive.TypicalInputs.STIMULUS).setActivation(1);
+				//Specify the stimulus for the AUTONOMY drive.
+				CombinedInputSpace.get(Personality.PrimaryDrives.AUTONOMY)
+					.get(Drive.TypicalInputs.STIMULUS).setActivation(0);
+				System.out.print("You say, \"Hello\". ");
+			}
+			else
+			{
+				//Specify the salutation to give as "Goodbye".
+				CombinedInputSpace.get("Salutations").get("Hello").setActivation(0);
+				CombinedInputSpace.get("Salutations").get("Goodbye").setActivation(1);
+				//Specify the stimulus for the AFFILIATON_AND BELONGINNESS drive.
+				CombinedInputSpace.get(Personality.PrimaryDrives.AFFILIATION_AND_BELONGINGNESS)
+					.get(Drive.TypicalInputs.STIMULUS).setActivation(0);
+				//Specify the stimulus for the AUTONOMY drive.
+				CombinedInputSpace.get(Personality.PrimaryDrives.AUTONOMY)
+					.get(Drive.TypicalInputs.STIMULUS).setActivation(1);
+				System.out.print("You say, \"Goodbye\". ");
+			}
+			if(i == 0)
+			{
+				/*Have the agent perceive the world. We are at trial 0, so no feedback has been 
+				 * provided yet.*/
+				Agent.perceive(CombinedInputSpace);
+			}
+			else
+			{
+				/*Have the agent perceive the world including the feedback for the last 
+				action it performed.*/
+				Agent.perceive(CombinedInputSpace, feedback);
+			}
+			//Have the agent choose an action.
+			ChosenAction = Agent.act();
+			if(ChosenAction.equalsID(Actions.get("Hello")))
+			{
+				//The agent said "Hello".
+				System.out.print("The agent said, \"Hello\". ");
+				if(CombinedInputSpace.get("Salutations").get("Hello").getActivation() == 1.0)
+				{
+					//The agent was right.
+					System.out.println("The agent was correct.");
+					//Record the agent's success.
+					CorrectCounter++;
+					//Give positive feedback.
+					feedback = 1;
+				}
+				else
+				{
+					//The agent was wrong.
+					System.out.println("The agent was incorrect.");
+					//Give negative feedback.
+					feedback = 0;
+				}
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 299:319, 320:340.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.